RecoLocalCalo/HGCalRecProducers/plugins/HGCalMultiClusterProducer.cc

```diff
@@ -0,0 +1,111 @@
+#ifndef __RecoLocalCalo_HGCRecProducers_HGCalMultiClusterProducer_H__
+#define __RecoLocalCalo_HGCRecProducers_HGCalMultiClusterProducer_H__
+
+// user include files
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/stream/EDProducer.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+
+#include "RecoParticleFlow/PFClusterProducer/interface/RecHitTopologicalCleanerBase.h"
+#include "RecoParticleFlow/PFClusterProducer/interface/SeedFinderBase.h"
+#include "RecoParticleFlow/PFClusterProducer/interface/InitialClusteringStepBase.h"
+#include "RecoParticleFlow/PFClusterProducer/interface/PFClusterBuilderBase.h"
+#include "RecoParticleFlow/PFClusterProducer/interface/PFCPositionCalculatorBase.h"
+#include "RecoParticleFlow/PFClusterProducer/interface/PFClusterEnergyCorrectorBase.h"
+
+#include <memory>
+#include <chrono>
+
+#include "RecoLocalCalo/HGCalRecAlgos/interface/HGCalImagingAlgo.h"
+#include "RecoLocalCalo/HGCalRecAlgos/interface/HGCalDepthPreClusterer.h"
+#include "RecoLocalCalo/HGCalRecAlgos/interface/HGCal3DClustering.h"
+
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "Geometry/Records/interface/IdealGeometryRecord.h"
+#include "Geometry/HGCalGeometry/interface/HGCalGeometry.h"
+
+#include "DataFormats/ParticleFlowReco/interface/PFCluster.h"
+
+class HGCalMultiClusterProducer : public edm::stream::EDProducer<> {
```

Comment:
could this be a global producer? It doesn't look like it modifies any member variables in the `produce()` function

Reply:
in the medium term, for GPU implementation, we will need it to stay stream producer

Reply:
Ah, okay.